tests/appsec/test_processor.py

```diff
@@ -86,11 +52,28 @@ def test_retain_traces(tracer, appsec):
     assert span.context.sampling_priority == priority.USER_KEEP
 
 
-def test_valid_json(tracer, appsec):
-    appsec.enable(tracer)
+def test_valid_json(tracer):
+    tracer._initialize_span_processors(appsec_enabled=True)
 
     with tracer.trace("test", span_type="web") as span:
         span.set_tag("http.url", "http://example.com/.git")
         span.set_tag("http.status_code", "404")
 
     assert "triggers" in json.loads(span.get_tag("_dd.appsec.json"))
+
+
+def test_enable_import_failure(tracer):
```

Comment:
A subprocess test case might be better suited here rather than trying to depend on `sys.modules` semantics:

https://github.com/DataDog/dd-trace-py/blob/d6d1705c1326ab74508a691d95f16b3f4187c006/tests/integration/test_integration.py#L637-L644

Reply:
Ack, will do in a further round